wgpu-hal/src/gles/egl.rs

```diff
@@ -550,21 +550,25 @@ impl Inner {
         let supports_khr_context = display_extensions.contains("EGL_KHR_create_context");
 
         //TODO: make it so `Device` == EGL Context
-        let mut context_attributes = vec![
-            khronos_egl::CONTEXT_MAJOR_VERSION,
-            3, // Request GLES 3.0 or higher
-        ];
-
-        if force_gles_minor_version != wgt::Gles3MinorVersion::Automatic {
+        let mut context_attributes = vec![];
+        if !supports_opengl {
+            context_attributes.push(khronos_egl::CONTEXT_MAJOR_VERSION);
+            context_attributes.push(3); // Request GLES 3.0 or higher
+            if force_gles_minor_version != wgt::Gles3MinorVersion::Automatic {
+                context_attributes.push(khronos_egl::CONTEXT_MINOR_VERSION);
+                context_attributes.push(match force_gles_minor_version {
+                    wgt::Gles3MinorVersion::Version0 => 0,
+                    wgt::Gles3MinorVersion::Version1 => 1,
+                    wgt::Gles3MinorVersion::Version2 => 2,
+                    _ => unreachable!(),
+                });
+            }
+        } else {
+            context_attributes.push(khronos_egl::CONTEXT_MAJOR_VERSION);
+            context_attributes.push(3);
             context_attributes.push(khronos_egl::CONTEXT_MINOR_VERSION);
-            context_attributes.push(match force_gles_minor_version {
-                wgt::Gles3MinorVersion::Version0 => 0,
-                wgt::Gles3MinorVersion::Version1 => 1,
-                wgt::Gles3MinorVersion::Version2 => 2,
-                _ => unreachable!(),
-            });
+            context_attributes.push(3);
```

Comment:
I think we should clarify on the user facing `InstanceDescriptor::gles_minor_version` that it is ignored when desktop OpenGL is chosen.
I'm also still pondering if this should file a warning if we're in OpenGL mode and and this is not set to automatic 🤔, might be annoying though.

Reply:
What would be the intention when forcing the minor version?

Theoretically we could also use this to force an OpenGL ES backend, but this behavior would differ from what WGL is doing (as it never uses OpenGL ES anyway)

Reply:
I don't know for sure but the docs imply to me that this was used as a workaround in Angle at some point or still is 🤔  https://github.com/gfx-rs/wgpu/blob/trunk/wgpu-types/src/lib.rs#L7071
Forcing a GLES backend might be useful in a similar vain, but if we want that we should take this separately :)